src/home_robot/home_robot/navigation_planner/discrete_planner.py

```diff
@@ -283,24 +303,67 @@ def plan(
                 self.min_goal_distance_cm,
             )
 
+            m_relative_stg_x, m_relative_stg_y = [
+                CM_TO_METERS * self.map_resolution * d
+                for d in [relative_stg_x, relative_stg_y]
+            ]
+            print("continuous actions for exploring")
+            print("agent angle =", angle_agent)
+            print("angle stg goal =", angle_st_goal)
+            print("angle final goal =", relative_angle_goal)
+            print(m_relative_stg_x, m_relative_stg_y, "rel ang =", relative_angle)
+            print("-----------------")
+
         # Short-term goal -> deterministic local policy
         if not (found_goal and stop):
-            if relative_angle > self.turn_angle / 2.0:
-                action = DiscreteNavigationAction.TURN_RIGHT
-            elif relative_angle < -self.turn_angle / 2.0:
-                action = DiscreteNavigationAction.TURN_LEFT
+            if self.discrete_actions:
+                if relative_angle > self.turn_angle / 2.0:
+                    action = DiscreteNavigationAction.TURN_RIGHT
+                elif relative_angle < -self.turn_angle / 2.0:
+                    action = DiscreteNavigationAction.TURN_LEFT
+                else:
+                    action = DiscreteNavigationAction.MOVE_FORWARD
             else:
-                action = DiscreteNavigationAction.MOVE_FORWARD
+                # Use the short-term goal to set where we should be heading next
+                m_relative_stg_x, m_relative_stg_y = [
+                    CM_TO_METERS * self.map_resolution * d
+                    for d in [relative_stg_x, relative_stg_y]
+                ]
+                if np.abs(relative_angle) > self.turn_angle / 2.0:
+                    # Must return commands in radians and meters
+                    relative_angle = math.radians(relative_angle)
+                    action = ContinuousNavigationAction([0, 0, -relative_angle])
+                else:
+                    # Must return commands in radians and meters
+                    relative_angle = math.radians(relative_angle)
+                    # relative_angle_goal = math.radians(relative_angle_goal)
+                    # action = ContinuousNavigationAction([m_relative_stg_y, m_relative_stg_x, -relative_angle])
+                    xyt_global = [m_relative_stg_y, m_relative_stg_x, -relative_angle]
```

Comment:
So the agent can move to a point not directly in front of it? Currently, this is not supported in the teleport action space we are using in the sim. For supporting continuous actions, I will try to do this in 3 sim steps. 1. turn towards waypoint, 2. move forward, 3. turn to reach the target yaw

Reply:
While it is easy to support "teleport to [x, y, yaw]", the collision check would be tricky as the real-world controller would not necessarily take a linear path.

Reply:
Got it. Yeah, the waypoint will be within `turn_angle/2`. So at first, I can try projecting it to the agent's forward direction

Reply:
Forward direction seems ok for simulation